crates/astria-conductor/src/executor.rs

```diff
@@ -259,8 +299,19 @@ impl<C: ExecutionClientV1Alpha1> Executor<C> {
             .cloned();
         match maybe_execution_block_hash {
             Some(execution_block_hash) => {
-                self.finalize_block(execution_block_hash, sequencer_block_hash)
+                // this case means block has already been executed.
+                // setting execution chain's FIRM and SOFT as this block
+                let executed_block = Block {
+                    number: block.header.height.value() as u32,
+                    hash: execution_block_hash.clone(),
+                    parent_block_hash: self.commitment_state.firm.clone().unwrap().hash,
+                    timestamp: Some(convert_tendermint_to_prost_timestamp(block.header.time)?),
+                };
+                self.update_commitment_state(executed_block.clone(), executed_block)
```

Comment:
The second `execute_block` should be replaced with `self.commitment_state.soft.unwrap().hash`.
It's likely that the height of the `soft` commit will be ahead of the `firm` commit.

Reply:
I don't think that's correct.
See: https://github.com/astriaorg/astria/pull/354#discussion_r1327953266

Reply:
This statement should be swapped with the one on line 323. See comment below.